engine/src/test/java/io/camunda/zeebe/streamprocessor/StreamProcessorTest.java

```diff
@@ -286,6 +286,9 @@ public void shouldSetSourcePointerForFollowUpRecords() {
     final var firstRecord = logStreamReader.next();
     assertThat(firstRecord.getSourceEventPosition()).isEqualTo(-1);
     final var firstRecordPosition = firstRecord.getPosition();
+
+    await("should write follow up events")
+        .untilAsserted(() -> assertThat(logStreamReader.hasNext()).isTrue());
     while (logStreamReader.hasNext()) {
       final var followUpRecord = logStreamReader.next();
       assertThat(followUpRecord.getSourceEventPosition()).isEqualTo(firstRecordPosition);
```

Comment:
:wrench: Since we are expecting exactly two follow up records, I suggest to remove the loop and verify that.
```
    await("should write follow up events")
        .untilAsserted(() -> assertThat(logStreamReader.hasNext()).isTrue());    
    assertThat(logStreamReader.next().getSourceEventPosition()).isEqualTo(firstRecordPosition);
    assertThat(logStreamReader.next().getSourceEventPosition()).isEqualTo(firstRecordPosition);
```